### PR TITLE
KOGITO-4341 Infinispan upgrade to version 12

### PR DESCRIPTION
**Pull request by cristianonicolai**
Assembly:
- https://github.com/kiegroup/kogito-runtimes/pull/1031
- https://github.com/kiegroup/kogito-runtimes/pull/1341
- https://github.com/kiegroup/kogito-apps/pull/815
- https://github.com/kiegroup/kogito-examples/pull/692

### CODE REVIEW

**Comment by cristianonicolai** [comment]
jenkins please retest this

**Comment by radtriste** [comment]
@cristianonicolai is it duplicate from https://github.com/kiegroup/kogito-apps/pull/846 ?

**Comment by cristianonicolai** [comment]
jenkins please retest this

**Comment by cristianonicolai** [comment]
jenkins please retest this

**Review by nmirasch — APPROVED** [review]

**Comment by cristianonicolai** [comment]
jenkins please retest this

**Comment by cristianonicolai** [comment]
@radtriste green! :) please review

File: data-index/data-index-service/data-index-service-common/src/test/java/org/kie/kogito/index/service/AbstractDomainIndexingServiceIT.java [hunk 1]
```diff
@@ -115,13 +115,14 @@ void testAddProtoFileMissingModelType() {
...
             assertThat(ex.getMessage()).isEqualTo(
-                    "Could not find message with name: traveller in proto file, e, please review option kogito_model");
+                    "Could not find message with name: org.demo.traveller in proto file, e, please review option kogito_model");
         }
     }
 
     @Test //Reproducer for KOGITO-172
     void testAddProtoFileTwice() throws Exception {
         protobufService.registerProtoBufferType(getProtoBufferFileV1());
+        Thread.sleep(10 * 1000);
```
Comment (radtriste) [inline data-index/data-index-service/data-index-service-common/src/test/java/org/kie/kogito/index/service/AbstractDomainIndexingServiceIT.java]: Do we really want that in tests ?

**Review by radtriste — APPROVED** [review]
one comment :)

File: data-index/data-index-service/data-index-service-common/src/test/java/org/kie/kogito/index/service/AbstractDomainIndexingServiceIT.java (hunk 1 above)
Comment (cristianonicolai) [reply data-index/data-index-service/data-index-service-common/src/test/java/org/kie/kogito/index/service/AbstractDomainIndexingServiceIT.java]: not really :)

**Pull request merged**